Framework/Algorithms/src/ConvertUnits.cpp

```diff
@@ -122,15 +155,15 @@ void ConvertUnits::exec() {
     }
   }
 
-  if (inputWS->x(0).size() < 2) {
+  if (correctWS->x(0).size() < 2) {
```

Comment:
I think the error message below is enough? It's because when the workspace is holding `BinEdges`, having 1 value doesn't makes sense, because the first would be the start of the bin and the second the end. If there's a single value it would be holding `Points`.